plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -2661,6 +2702,8 @@ void producer_plugin_impl::schedule_production_loop() {
    } else {
       fc_dlog(_log, "Speculative Block Created");
    }
+
+   _time_tracker.start_idle_time();
```

Comment:
Hum, I wonder if it shouldn't be `add_idle_time`, with the change I mentioned before in `add_idle_time` resetting `last_trx_time_point = now; `

Reply:
I don't understand; there is no idle time to add.

Reply:
OK, I see that `start_idle_time();` does:

`last_trx_time_point = now;`

So assuming that `now` != `last_trx_time_point`, what was the time spent between `last_trx_time_point` and `now` used for?

Reply:
It was used for start block which includes potentially many trx processing.

Reply:
So create a `trx_time_tracker` and immediately `cancel()` it?

Reply:
I realize this is not counting the time for trxs in start block. I'll fix that.

Reply:
> It was used for start block which includes potentially many trx processing.

If we processed trx, they should have updated `last_trx_time_point` at the end.

Reply:
> So create a `trx_time_tracker` and immediately `cancel()` it?

No, just call `add_idle_time()` instead of `start_idle_time()`

Reply:
But that is not idle time. It is time spent creating a pending block state, activating protocol features, removing expired trxs, removing expired blacklisted trxs, reposting exhausted read-only trxs. Also potentially includes processing trxs but then and only then is the idle time reset.

Reply:
So you want it to go to "other". If so then I'm OK with it. 

I think it would be better to have a counter for "other" (as we have for fail/success/idle) so we can check that `fail + success + idle + other` == `wall time`, but maybe this could be done later.

Reply:
It would require many additions of `add_other_time` as we would need to do that in many places in start_block and abort_block.

Reply:
Not really, it would be replacing `start_idle_time()` with `add_other_time()` I think.

Reply:
Would require `add_other_time()` before `process_unapplied_trxs`,  and `process_incoming_trxs` and once in `abort_block` so not too bad. I'll try adding it.

Reply:
We create a `trx_time_tracker` only when we start processing a transaction, right? So could it do `add_other_time` in its constructor?

Reply:
BTW I really like the RAII implementation!

Reply:
> We create a `trx_time_tracker` only when we start processing a transaction, right? So could it do `add_other_time` in its constructor?

But then you are just back to what it does now except you explicitly track the value instead of just calculate it.

Reply:
Hum, it is kind of true, especially with the current version of the code which increases the time_point every time we add a duration. 
But still, if we track the "other" value explicitly, we can verify that the sum of all accounted times is the elapsed time, which may be not true if we were to create `trx_time_tracker` objects on more than one thread for example.

Reply:
Added explicit tracking of other. Also added a pause/unpause for when in read-window as it didn't seem correct to call that "other" time.